controllers/pipeline/pipeline_adapter.go

```diff
@@ -67,6 +67,11 @@ func (a *Adapter) EnsureSnapshotExists() (results.OperationResult, error) {
 		return results.ContinueProcessing()
 	}
 
+	if a.component == nil {
```

Comment:
I am just learning to debug it from you, this is to stop the later process because a component is not associated because of some unknown reason, right? And what the user will see for this type of non-associated component?

Reply:
Yes, basically this means that the build pipeline doesn't have a component associated with it (by way of having a label that specifies it). In such cases it is not possible to construct a new Snapshot in this function so we output a log message and ignore that pipeline.

Reply:
I see now, thank you!